File: src/main/java/entry/SQLStructBuilderTest.java
package entry;

import org.junit.Test;

import static org.junit.Assert.*;

public class SQLStructBuilderTest {

    Builder builder = new SQLStructBuilder();

    @Test
    public void gen() {
        String s = "CREATE TABLE IF NOT EXISTS `live_product`\n" +
                "(\n" +
                "    `id`         INT(11)     NOT NULL AUTO_INCREMENT,\n" +
                "    `room_id`    VARCHAR(32) NOT NULL COMMENT '房间id',\n" +
                "    `live_id`    VARCHAR(32) NOT NULL COMMENT '直播id',\n" +
                "\n" +
                "    `product_id` INT(11)     NOT NULL,\n" +
                "    `pos`        INT(11)     NOT NULL COMMENT '排序值，从0开始，越小越靠前',\n" +
                "\n" +
                "    `status`     INT(11)     NOT NULL DEFAULT 1 COMMENT '1: 正常，2：已删除',\n" +
                "    `create_at`  DATETIME    NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'create time',\n" +
                "    `update_at`  DATETIME    NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT '更新时间',\n" +
                "\n" +
                "    KEY `idx` (room_id),\n" +
                "    PRIMARY KEY (`id`),\n" +
                "    UNIQUE KEY `idx_room_id_live_id` (`room_id`, `live_id`, `product_id`)\n" +
                ") ENGINE = INNODB\n" +
                "  CHARSET = utf8mb4 COMMENT '直播商品列表';";
        System.out.println(builder.gen(s));
    }
}

File: src/main/kotlin/entry/SQLStructBuilderTest.kt
package entry

import org.junit.Test
import kotlin.test.assertEquals

class SQLStructBuilderTest {

    @Test
    fun gen() {
        val case1 = """
            CREATE TABLE `policy_info` (
            `id` bigint(11) unsigned NOT NULL AUTO_INCREMENT COMMENT ' comment id',
            `policy_id` varchar(128) NOT NULL COMMENT 'ID',
            `policy_name` varchar(128) NOT NULL,
            `object_id` varchar(128) NOT NULL COMMENT 'ID',
            `policy_type` varchar(16) NOT NULL DEFAULT 'system' COMMENT 'policy_type="system":policy_type="custom":',
            `policy_version` int(11) NOT NULL COMMENT 'version=1',
            `statement` mediumtext NOT NULL COMMENT 'json',
            `is_delete` tinyint(4) NOT NULL DEFAULT '0' COMMENT 'is_delete=0:is_delete=1:
            
            ',
            `creator` varchar(128) NOT NULL,
            `operator` varchar(128) NOT NULL,
            `create_time` datetime NOT NULL,
            `update_time` datetime NOT NULL,
            PRIMARY KEY (`id`),
            UNIQUE KEY `uk_policy_id` (`policy_id`),
            KEY `idx_object_id` (`object_id`)
            ) ENGINE=InnoDB AUTO_INCREMENT=11 DEFAULT CHARSET=utf8mb4;
        """.trimIndent()
        val builder = SQLStructBuilder()
        builder.setConfig("", false)
        assertEquals(builder.gen(case1), """type PolicyInfo struct {
	ID	uint64 //  comment id
	PolicyID	string // ID
	PolicyName	string
	ObjectID	string // ID
	PolicyType	string // policy_type=system:policy_type=custom:
	PolicyVersion	int // version=1
	Statement	string // json
	IsDelete	int8 // is_delete=0:is_delete=1:  
	Creator	string
	Operator	string
	CreateTime	time.Time
	UpdateTime	time.Time
}

func (m *PolicyInfo) TableName() string {
	return "policy_info"
}""")
    }
}
